File: Medii de Proiectare si Programare/HW/mpp_hw/src/main/java/mpp/com/Service/ServiceChild.java
package mpp.com.Service;

import mpp.com.Domain.Child;
import mpp.com.Domain.ChildTrackDTO;
import mpp.com.Domain.Track;
import mpp.com.Repository.RepoDBChild;

import java.util.List;
import java.util.Optional;

public class ServiceChild {

    RepoDBChild repoDBChild;

    public ServiceChild(RepoDBChild repoDBChild) {
        this.repoDBChild = repoDBChild;
    }

    public Optional<Child> searchChildById(Long id) throws IllegalArgumentException {
        return repoDBChild.findOne(id);
    }

    public Iterable<Child> getAllChildren() {
        return repoDBChild.findAll();
    }

    public void addChild(String name, int age, List<Track> tracks) {

        var child = new Child(name, age);
        child.setTracks(tracks);
        repoDBChild.save(child);
    }

    public Iterable<ChildTrackDTO> getChildrenByTracks(Long idTrack) {
        return repoDBChild.getChildrenByTracks(idTrack);
    }

    public Iterable<Child> getChildrenByTrackId(Long idTrack) {
        return repoDBChild.getChildrenByTrackId(idTrack);
    }
}

File: Medii de Proiectare si Programare/HW/MPP_CSharp/MPP_CSharp/Service/ServiceChild.cs
using System;
using System.Collections.Generic;
using MPP_CSharp.Domain;
using MPP_CSharp.Repository;

namespace MPP_CSharp.Service
{
    public class ServiceChild
    {
        private readonly IRepositoryChild _repoDBChild;

        public ServiceChild(IRepositoryChild repoDBChild)
        {
            _repoDBChild = repoDBChild ?? throw new ArgumentNullException(nameof(repoDBChild));
        }

        public Child? SearchChildById(long id)
        {
            return _repoDBChild.Search(id);
        }

        public IEnumerable<Child> GetAllChildren()
        {
            return _repoDBChild.GetAll();
        }

        public void AddChild(string name, int age, List<Track> tracks)
        {
            var child = new Child(name, age) { Tracks = tracks };
            _repoDBChild.Save(child);
        }

        public IEnumerable<ChildTrackDTO> GetChildrenByTracks(long idTrack)
        {
            return _repoDBChild.GetChildrenByTracks(idTrack);
        }

        public IEnumerable<Child> GetChildrenByTrackId(long idTrack)
        {
            return _repoDBChild.GetChildrenByTrackId(idTrack);
        }
    }
}